### PR TITLE
documentation: Ocean State Estimation Packages (converted from tex)

### PR DESCRIPTION
**Pull request by gaelforget**
Was originally parsed through pandoc (pandoc -s -f latex -t rst -o top_section.rst top_section.tex); webhooked to read the docs (https://mitgcm-via-hubgcm.readthedocs.io); then various issues were fixed (tables, cross-referencing, etc.).

### CODE REVIEW

File: doc/ocean_state_est/ocean_state_est.rst [hunk 1]
```diff
@@ -1,3 +1,1284 @@
 Ocean State Estimation Packages
-*******************************
+===============================
 
+This chapter describes packages that have been introduced for ocean
+state estimation purposes and in relation with automatic differentiation
+(see :ref:`chap_autodiff`)
+
+.. FYI the corresponding tex was: (see Chapter~\ref{chap:autodiff})
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst]: These comments don't need to stay long term.

File: doc/ocean_state_est/ocean_state_est.rst [hunk 2]
```diff
@@ -1,3 +1,1284 @@
...
+ECCO: model-data comparisons using gridded data sets
+----------------------------------------------------
+
+The functionalities implemented in ``pkg/ecco`` are: (1) output
+time-averaged model fields to compare with gridded data sets; (2)
+compute normalized model-data distances (i.e., cost functions); (3)
+compute averages and transports (i.e., integrals). The former is
+achieved as the model runs forwards in time whereas the others occur
+after time-integration has completed. Following
+:raw-latex:`\cite{for-eta:15}` the total cost function is formulated
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst]: This :raw-latex: directive should be replaced a with :cite: directive. `for-eta:15` may also need to be added to the bib file.

File: doc/ocean_state_est/ocean_state_est.rst [hunk 3]
```diff
@@ -1,3 +1,1284 @@
...
+generically as
+
+.. FYI the equations below were translated by pandoc as:
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst]: More comments that don't need to stay long term.

File: doc/ocean_state_est/ocean_state_est.rst [hunk 4]
```diff
@@ -1,3 +1,1284 @@
...
+  | :math:`\mathcal{D}`               | diagnostic computation operator   |
+  +-----------------------------------+-----------------------------------+
+  | :math:`\mathcal{S}`               | averaging/subsampling operator    |
+  +-----------------------------------+-----------------------------------+
+  | :math:`\mathcal{Q}`               | Pre-processing operator           |
+  +-----------------------------------+-----------------------------------+
+  | :math:`\mathcal{R}`               | Pre-conditioning operator         |
+  +-----------------------------------+-----------------------------------+
+
+.. FYI the above table was mis-handled by pandoc which had
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst]: More comments

File: doc/ocean_state_est/ocean_state_est.rst [hunk 5]
```diff
@@ -1,3 +1,1284 @@
...
+  |                       |                       | :math:`\vec{o}_i` and             |
+  |                       |                       | :math:`\vec{m}_i`                 |
+  |                       |                       | (see text)                        |
+  +-----------------------+-----------------------+-----------------------------------+
+  |``gencost_outputlevel``| integer               | Greater than 0 will               |
+  |                       |                       | output misfit fields              |
+  +-----------------------+-----------------------+-----------------------------------+
+  | ``gencost_errfile``   | character(*)          | Uncertainty field                 |
+  |                       |                       | name (not used in                 |
+  |                       |                       | section \ `1.2 <#intg             |
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst]: This (and following) section references would be simpler if they used the :numref: directive. See http://mitgcm-example.readthedocs.io/en/latest/contributing/contributing.html#internal-document-references for examples of the syntax to use.

File: doc/ocean_state_est/ocean_state_est.rst [hunk 6]
```diff
@@ -1,3 +1,1284 @@
...
+
+When the k2 option is used (e.g. for cubed sphere runs), the input file
+is to be completed with interpolation grid points and coefficients
+computed offline using netcdf_ecco_GenericgridMain.m. Typically, you
+would first provide the standard namelist and files. After detecting
+that interpolation information is missing, the model will generate
+special grid files (profilesXCincl1PointOverlap\* etc.) and then stop.
+You then want to run netcdf_ecco_GenericgridMain.m using the special
+grid files. *This operation could eventually be inlined.*
+
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst:584]: It is possible to have large sections of verbatim text by placing two colons followed by a blank line and then indenting the text.

```
::

   verbatim text
   more verbatim text
```

File: doc/ocean_state_est/ocean_state_est.rst [hunk 7]
```diff
@@ -1,3 +1,1284 @@
...
+this operator is specified by ``gen*Precond`` (:math:`=1` by default).
+
+.. _sec:pkg:smooth:
+
+SMOOTH: Smoothing And Covariance Model
+--------------------------------------
+
+TBA ...
+
+The line search optimisation algorithm [sectionoptim]
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst]: The label for this section seems to have been incorporated into the title.

File: doc/ocean_state_est/ocean_state_est.rst [hunk 8]
```diff
@@ -1,3 +1,1284 @@
...
+
+OPWARMI, OPWARMD files
+^^^^^^^^^^^^^^^^^^^^^^
+
+Two files retain values of previous iterations which are used in latest
+iteration to update Hessian:
+
+-  **OPWARMI**: contains index settings and scalar variables
+
+   .. raw:: latex
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst]: This :raw-latex: directive and the others soon afterwards aren't doing anything.

File: doc/ocean_state_est/ocean_state_est.rst [hunk 9]
```diff
@@ -1,3 +1,1284 @@
...
+          |      (---- CALL LSUPDXX
+          |      )       |
+          |      (       |---- >>> if jmax=0 <<<
+          |      )       |      |  first optimization after cold start:
+          |      (       |      |  preconditioner estimated via ff_0 - ff_(first guess)
+          |      )       |      |  dd(i-1) = -gg(i-1)*preco
+          |      (       |      |  
+          |      )       |     >>> if jmax > 0 <<<
+          |      (       |         dd(i-1) = -gg(i-1)
+          |      )
```
Comment (edoddridge) [inline doc/ocean_state_est/ocean_state_est.rst]: This isn't a very helpful footnote. Can we clarify it somehow?

**Review by edoddridge — CHANGES_REQUESTED** [review]
Thanks @gaelforget! This is a great start.

There are a few issues that we need to address before merging this PR. The most pressing is reducing the number of commits, probably to a single commit, and improving the commit message. I think [this](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit) is a fairly good guide on how to squash commits together. Let me know if you'd like more guidance - I'm happy to come to the 14th floor and talk you through it.

I've gone through and highlight some places where I think we can improve on this before we merge. 

Thanks again for helping with the translation, and for testing the GitHub workflow.

**Comment by gaelforget** [comment]
Should be good to go now?

File: doc/ocean_state_est/ocean_state_est.rst (hunk 6 above)
Comment (gaelforget) [reply doc/ocean_state_est/ocean_state_est.rst:584]: Thanks

**Comment by edoddridge** [comment]
I think it's very close.

- The section references automatically get "Section" in front of the number, so there are a few spots where the text now has "section Section".
- The footnote "Not sure what this means." is still there. It's not a particularly helpful footnote for a manual to have...

**Comment by gaelforget** [comment]
Done.

**Review by edoddridge — APPROVED** [review]

**Comment by edoddridge** [comment]
LGTM

**Comment by christophernhill** [comment]
Awesome - and thanks Gael! 

merging now

**Pull request merged**